src/components/Upload/Uploader.tsx

```diff
@@ -214,8 +214,43 @@ export default function Uploader(props: Props) {
             addLogLine(`received file upload request`);
             setWebFiles(props.webFileSelectorFiles);
         } else if (props.dragAndDropFiles?.length > 0) {
-            addLogLine(`received drag and drop upload request`);
-            setWebFiles(props.dragAndDropFiles);
+            if (isElectron()) {
+                const main = async () => {
+                    try {
+                        addLogLine(`uploading dropped files from desktop app`);
+                        // check and parse dropped files which are zip files
+                        const electronFiles = [
+                            ...(await Promise.all(
+                                props.dragAndDropFiles.map(async (file) => {
+                                    if (file.name.endsWith('.zip')) {
+                                        const zipFiles =
+                                            await importService.getElectronFilesFromGoogleZip(
+                                                (file as any).path
+                                            );
+                                        addLogLine(
+                                            `zip file - ${file.name} contains ${zipFiles.length} files`
+                                        );
+                                        return zipFiles;
+                                    } else {
+                                        return file as unknown as ElectronFile;
```

Comment:
What's the difference between.. Can we add a comment why we have to cast it as `unknown` first?
`return file  as ElectronFile;` vs `return file as unknown as ElectronFile`;

Reply:
We earlier created a separate type for files coming from Electron(desktop), but mostly they can be used interchangeably.

I didn't want to do the complete type and make this PR too big and bloated with non-related changes.

So, for now, I did `file as unknown as ElectronFile;` which just is a way to tell the compiler to ignore the type difference between the two types and type cast file as ElectronFile.


Yes adding a comment